esp/public/media/scripts/program/modules/adminclass.js

```diff
@@ -247,45 +247,46 @@ function createClassRow(clsObj)
 ";
 
     // Now fill in the values in the template
-    var teacher_list = clsObj.teachers;
-    teacher_list = $j.map(teacher_list, function(val, index) {
-	return json_data.teachers[val].first_name + " " + json_data.teachers[val].last_name;
-    });
-    var teacher_list_string = teacher_list.join(", ");
+  var teacher_list = clsObj.teachers;
 
-    var section_link_list = "";
-    for (var i = 0; i < clsObj.sections.length; i++)
-    {
-	var section = sections[clsObj.sections[i]];
-	section_link_list = section_link_list.concat("<a href='/teach/"+base_url+"/select_students/"+section.id+"'>Sec. "+section.index+"</a><br />");
-    }
+  var teacher_link_list = "";
+  for (var i in teacher_list) {
+      var teacher = json_data.teachers[teacher_list[i]];
+      var teacherName = teacher.first_name + " " + teacher.last_name;
+      teacher_link_list = teacher_link_list.concat('<a href="/manage/userview?username=' + teacher.username + '">' + teacherName + '</a><br />');
```

Comment:
Note to self or whomever ends up cleaning this up: here's another XSS issue (which may exist in the original too :/ ) .